domain/consensus/processes/consensusstatemanager/resolve.go

```diff
@@ -107,13 +183,14 @@ func (csm *consensusStateManager) ResolveVirtual(maxBlocksToResolve uint64) (*ex
 		return nil, false, err
 	}
 
+	// TODO: why was `readStagingArea` used here ?
```

Comment:
This looks like something temporary that you forgot to remove?

Reply:
I also wanted to understand it. How can the `readStagingArea` have the updated virtual info just written through `updateVirtualStagingArea` ? I mean this is how it was before I changed it and was wondering if it might be another bug. 
@someone235 @svarogg